app/assets/javascripts/discourse/app/routes/signup.js

```diff
@@ -35,6 +35,7 @@ export default class extends DiscourseRoute {
     const { referrer } = document;
     const { canSignUp } = this.controllerFor("application");
     const { isOnlyOneExternalLoginMethod, singleExternalLogin } = this.login;
+    const redirect = auth_immediately || login_required || !from || wantsTo;
```

Comment:
For a follow-up, it's a little unclear what the `!from` and `wantTo` properties here represent.